Support /version; fix some global URIs

jaymzh opened the pull request:
/version is a stripped down version of what you get with EC/CS/OSC. Useful for knowing what you are talking to.

jaymzh commented:
@chef/client-core 
there is no "group" for "tools maintainers" that I can find, so I'm CCing the client maintainers

lamont-granquist commented:
LGTM

odcinek commented:
:+1:

jaymzh commented:
(FTR, I'm not a 'tools' maintainer and thus don't have merge access here)

tyler-ball commented:
:+1: 

Is @jkeiser the maintainer of chef-zero?

jaymzh commented:
Maintainers are @stevendanna @jtimberman @lamont-granquist and @danielsdeleo according to MAINTAINERS.md in chef.

Got 1/4 so far. :)

stevendanna commented:
:+1: Thanks for this.  Let's not wait too long to merge or the CHANGELOG whitespace changes will be a pain.

danielsdeleo commented on spec/server_spec.rb:
```diff
@@ -33,6 +33,12 @@ def get_nodes(accepts)
+      def get_version
+        uri = URI(@server.url)
+        httpcall = Net::HTTP.new(uri.host, uri.port)
+        httpcall.get('/version', 'Accept' => 'application/json')
```
Looks like we're asking for `application/json`, but the server is going to send `text/plain`, yes? According to the RFC I guess this is okay:

> Note: HTTP/1.1 servers are allowed to return responses which are not acceptable according to the accept headers sent in the request. In some cases, this may even be preferable to sending a 406 response. User agents are encouraged to inspect the headers of an incoming response to determine if it is acceptable.

Though I think web machine is generally pretty strict about this and would return a 406...

danielsdeleo commented:
One comment about content types, but :+1: otherwise.

stevendanna commented:
Assigning to me to clean up the content-type stuff, rebase, and merge unless @jaymzh get's to it first.

jaymzh commented:
Sorry. I was on vacation when these comments came in and then just never saw them. I will take this.

jaymzh commented:
The tests pass, but I updating the spec seems best. I'll rebase and re-push today.

jaymzh commented:
OK, rebased, updated content-type in request (chef-zero requires you to accept application/json, so I just added text/plain), and squashed all the commits.

jaymzh commented:
OK that's dan and steven and Tyler and I who are good with this and I cleaned up the content-type thing. Who has commit access here?
